src/context.ml

```diff
@@ -101,6 +101,9 @@ type t =
   ; which_cache             : (string, Path.t option) Hashtbl.t
   }
 
+let equal x y = String.equal x.name y.name
+let hash t = String.hash t.name
```

Comment:
Just so it's clear what I meant: `let equal = (==)` works with this hash function just fine, and in maybe that's better. (not sure if you chose against it or I failed to deliver my point)

Reply:
I think the latter is fine. I forgot that the contexts have names. But I think using physical equality is pretty safe as well.